usaspending_api/search/v2/views/spending_by_geography.py

```diff
@@ -283,10 +316,81 @@ def district_results(self, state_lookup):
+
+        if bucket_count == 0:
+            return None
+
+        # Add 100 to make sure that we consider enough records in each shard for accurate results
+        group_by_agg_key = A("terms", field=self.agg_key, size=bucket_count, shard_size=bucket_count + 100)
+        sum_aggregations = get_scaled_sum_aggregations(self.obligation_column)
+        sum_field = sum_aggregations["sum_field"]
+
+        search.aggs.bucket("group_by_agg_key", group_by_agg_key).metric("sum_field", sum_field)
+
+        # Set size to 0 since we don't care about documents returned
+        search.update_from_dict({"size": 0})
+
+        return search
+
+    def build_elasticsearch_result(self, response: dict) -> Dict[str, dict]:
+        results = {}
+        geo_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
+        for bucket in geo_info_buckets:
+            geo_info = json.loads(bucket.get("key"))
+
+            if self.geo_layer == GeoLayer.STATE:
+                display_name = geo_info.get("state_name").title()
+                shape_code = geo_info.get("state_code").upper()
+            elif self.geo_layer == GeoLayer.COUNTY:
+                display_name = geo_info["county_name"].title()
+                shape_code = f"{geo_info['state_fips']}{geo_info['county_code']}"
+            else:
+                display_name = f"{geo_info['state_code']}-{geo_info['congressional_code']}".upper()
+                shape_code = f"{geo_info['state_fips']}{geo_info['congressional_code']}"
+
+            per_capita = None
+            aggregated_amount = int(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100")
+            population = int(geo_info["population"]) if geo_info["population"] else None
+            if population:
+                per_capita = (Decimal(aggregated_amount) / Decimal(population)).quantize(Decimal(".01"))
+
+            results[shape_code] = {
+                "shape_code": shape_code or None,
+                "aggregated_amount": aggregated_amount,
+                "display_name": display_name or None,
+                "population": population,
+                "per_capita": per_capita,
+            }
+
+        return results
+
+    def query_elasticsearch(self, filter_query: ES_Q) -> list:
+        logger.info("Using experimental Elasticsearch functionality for 'spending_by_geography'")
+
+        search = self.build_elasticsearch_search_with_aggregation(filter_query)
+        if search is None:
+            return []
+        response = search.handle_execute()
+        results_dict = self.build_elasticsearch_result(response.aggs.to_dict())
+
+        if self.geo_layer_filters:
+            filtered_shape_codes = set(self.geo_layer_filters) & set(results_dict.keys())
+            results = [results_dict[shape_code] for shape_code in filtered_shape_codes]
+        else:
+            results = results_dict.values()
+
+        return results
```

Comment:
The current postgres implementation has the results sorted alphabetically by state/shape code, is that something we care about maintaining with ES? Not really a blocker, just something that makes it easier to test

Reply:
Unless I overlooked it I did not see an explicit attempt to sort results with the Postgres implementation. Since sorting hurts performance and the frontend doesn't need it sort I elected not to sort the results. If you think it should be sorted I can make that change.

Reply:
Nah I don't think it matters.